ui/selectors/selectors.js

```diff
@@ -1757,6 +1762,25 @@ export function getCustomTokenAmount(state) {
   return state.appState.customTokenAmount;
 }
 
+export function getUpdatedAndSortedAccounts(state) {
+  const accounts = getMetaMaskAccountsOrdered(state);
+  const pinnedAccounts = getPinnedAccountsList(state);
+  accounts.forEach((account) => {
+    account.pinned = Boolean(pinnedAccounts?.includes(account.address));
+  });
+
+  const sortedSearchResults = accounts.slice().sort((a, b) => {
```

Comment:
why is the slice needed here if we are not actually reducing the size of the array?